### PR TITLE
PR : dark mode

### PR DESCRIPTION
**Pull request by LaolouRavon-Laforest**

### CODE REVIEW

**Review by thmarinho — CHANGES_REQUESTED** [review]
Will look at this when #30 will be merged

File: frontend/pages/project.tsx [hunk 1]
```diff
@@ -3,14 +3,14 @@ import { GetServerSideProps } from 'next';
 
 const Project: React.FC<{ projects: string[] }> = ({ projects }) => {
     return (
-        <>
+        <div className="gb-red-600 dark:gb-grey-500">
```
Comment (thmarinho) [inline frontend/pages/project.tsx:6]: gb ?

File: frontend/styles/globals.css [hunk 2]
```diff
@@ -9,3 +9,11 @@
...
   @apply bg-gray-200 m-0 p-0 text-xs sm:text-sm md:text-base lg:text-lg xl:text-xl 2xl:text-2xl;
 }
+
+.theme-switch:checked {
+  @apply transition-transform transform translate-x-7 duration-500;
+}
+
+.theme-switch:checked + .theme-switch-label {
+  @apply bg-blue-300;
+}
```
Comment (thmarinho) [inline frontend/styles/globals.css:19]: Not used

File: frontend/pages/component/RootLayout.tsx [hunk 3]
```diff
@@ -2,6 +2,27 @@ import React from 'react';
 import NavBar from './NavBar';
 import Head from 'next/head';
 
+export const toggleTheme = (): boolean => {
+    if (typeof window !== 'undefined') {
+        const newTheme = isDarkTheme() ? 'light' : 'dark';
+        localStorage.setItem('theme', newTheme);
+        document.documentElement.classList.toggle('dark', newTheme === 'dark');
```
Comment (thmarinho) [inline frontend/pages/component/RootLayout.tsx:9]: `document.documentElement` è.é

File: frontend/pages/component/RootLayout.tsx [hunk 4]
```diff
@@ -2,6 +2,27 @@ import React from 'react';
...
+    return false;
+};
+
+export function isDarkTheme() {
+    if (typeof window !== 'undefined') {
+        const savedTheme = localStorage.getItem('theme');
+        if (savedTheme === 'dark') {
+            return true;
+        }
+        return false;
```
Comment (thmarinho) [inline frontend/pages/component/RootLayout.tsx:21]:
```suggestion
        return savedTheme === 'dark'
```
?

File: frontend/tailwind.config.ts [hunk 5]
```diff
@@ -6,6 +6,7 @@ const config: Config = {
         './components/**/*.{js,ts,jsx,tsx,mdx}',
         './app/**/*.{js,ts,jsx,tsx,mdx}',
     ],
+    darkMode: 'class',
```
Comment (thmarinho) [inline frontend/tailwind.config.ts:9]: move this above `content`

File: frontend/pages/component/NavBar.tsx [hunk 6]
```diff
@@ -95,6 +118,21 @@ export default function NavBar() {
                                     ))}
                                 </ul>
                             </li>
+                            <li className="mt-auto -mx-2">
+                                <button onClick={mainToggleTheme}>
+                                    {darkTheme ? (
+                                        <div className="group flex gap-x-3 rounded-md p-2 text-sm font-semibold leading-6 text-indigo-200 hover:bg-indigo-700 hover:text-white">
+                                            <MoonIcon className="h-6 w-6 shrink-0 text-indigo-200 group-hover:text-white" aria-hidden="true" />
+                                            Dark Mode
+                                        </div>
```
Comment (thmarinho) [inline frontend/pages/component/NavBar.tsx:127]: Duplicated from below

File: frontend/pages/component/NavBar.tsx [hunk 7]
```diff
@@ -59,6 +67,21 @@ export default function NavBar() {
...
+                                                    {darkTheme ? (
+                                                        <div className="group flex gap-x-3 rounded-md p-2 text-sm font-semibold leading-6 text-indigo-200 hover:bg-indigo-700 hover:text-white">
+                                                            <MoonIcon className="h-6 w-6 shrink-0 text-indigo-200 group-hover:text-white" aria-hidden="true" />
+                                                            Dark Mode
+                                                        </div>
+                                                    ) : (
+                                                        <div className="group flex gap-x-3 rounded-md p-2 text-sm font-semibold leading-6 text-indigo-200 hover:bg-indigo-700 hover:text-white">
+                                                            <SunIcon className="h-6 w-6 shrink-0 text-indigo-200 group-hover:text-white" aria-hidden="true" />
+                                                            Light mode
+                                                        </div>
```
Comment (thmarinho) [inline frontend/pages/component/NavBar.tsx:81]: Duplicated from above

**Review by thmarinho — CHANGES_REQUESTED** [review]

File: frontend/pages/component/NavBar.tsx [hunk 8]
```diff
@@ -10,7 +10,11 @@ import {
     FolderIcon,
     XMarkIcon,
     ArrowLeftOnRectangleIcon,
+    MoonIcon,
+    SunIcon,
 } from '@heroicons/react/24/outline';
+import { toggleTheme } from './RootLayout';
+import { isDarkTheme } from './RootLayout';
```
Comment (alanschnegg) [inline frontend/pages/component/NavBar.tsx:17]: Make only one import

File: frontend/pages/component/NavBar.tsx [hunk 9]
```diff
@@ -19,7 +23,11 @@ const navigation = [
 
 export default function NavBar() {
     const [sidebarOpen, setSidebarOpen] = useState(false);
-
+    const [darkTheme, setDarkTheme] = useState(isDarkTheme());
```
Comment (alanschnegg) [inline frontend/pages/component/NavBar.tsx:26]: Why define an useState to save if theme dark is used when you already have a function for that ?
You even use this function to set the default value

File: frontend/pages/component/NavBar.tsx [hunk 10]
```diff
@@ -19,7 +23,11 @@ const navigation = [
...
+    const mainToggleTheme = () => {
+        const newTheme = toggleTheme();
+        setDarkTheme(newTheme);
+    };
```
Comment (alanschnegg) [inline frontend/pages/component/NavBar.tsx:30]: Remove this and the useState and just use 'toggleTheme' and 'isDarkTheme' from the import

**Review by alanschnegg — CHANGES_REQUESTED** [review]

**Comment by thmarinho** [comment]
Closing PR

**Pull request closed without merging**